lib/src/ruby/ruby.cc

```diff
@@ -18,10 +46,21 @@ namespace facter { namespace ruby {
         return true;
     }
 
-    void load_custom_facts(collection& facts, vector<string> const& paths)
+    void load_custom_facts(collection& facts, bool initialize_puppet, vector<string> const& paths)
     {
         module mod(facts, paths);
+        if (initialize_puppet) {
+            try {
+                eval(load_puppet);
```

Comment:
Might as well just keep it a `Facter.reset` followed by a `Facter.external_search_path(Puppet[:pluginfactsdest]) if ...` in the ruby code snippet.